src/lib/Components/FilterModal.tsx

```diff
@@ -0,0 +1,105 @@
+import { Box, Button, CloseIcon, Flex, Sans } from "@artsy/palette"
+import React from "react"
+import { LayoutAnimation, Modal as RNModal, TouchableWithoutFeedback, ViewProperties } from "react-native"
+import styled from "styled-components/native"
+
+interface ModalProps extends ViewProperties {
+  visible?: boolean
+  closeModal?: () => void
+}
+
+interface State {
+  isComponentMounted: boolean
+}
+
+const ModalBackgroundView = styled.View`
+  background-color: #00000099;
+  flex: 1;
+  flex-direction: column;
+`
+
+const ModalInnerView = styled.View<{ visible: boolean }>`
+  flex-direction: column;
+  width: 100%;
+  background-color: white;
+  height: ${({ visible }) => (visible ? "auto" : "0")};
+  padding: ${({ visible }) => (visible ? "20px" : "0")};
+  border-top-left-radius: 10px;
+  border-top-right-radius: 10px;
+`
+
+export class FilterModal extends React.Component<ModalProps, State> {
```

Comment:
I don't know if we have a rule in Emission, but if this were in Reaction I'd expect this new component to be a functional component that uses hooks in place of the `isComponentMounted` bits.

Reply:
This component is going to eventually have to manage several states and transitions (given the various possible filter states) and for this reason, I wonder if it makes more sense to keep it as a class-based component to take advantage of the built-in class lifecycle methods.